src/ChainSafe.Gaming.WalletConnect/Wallets/WalletRegistry.cs

```diff
@@ -0,0 +1,107 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ChainSafe.Gaming.WalletConnect.Models;
+using ChainSafe.Gaming.Web3.Core;
+using ChainSafe.Gaming.Web3.Environment;
+using WalletConnectSharp.Common.Logging;
+
+namespace ChainSafe.Gaming.WalletConnect.Wallets
+{
+    /// <summary>
+    /// Default implementation for the registry of the wallets supported by WalletConnect.
+    /// </summary>
+    public class WalletRegistry : ILifecycleParticipant, IWalletRegistry
```

Comment:
Why is this a service? It doesn't seem like something that'll be swapped out

Reply:
This will probably need a Stub implementation for some tests. Maybe not, not sure on this one.

Reply:
It's unlikely that it will need one, it's just adding more parts imo

Reply:
Tried removing `IWalletRegistry`, but it made Unity go into an infinite loop. This is probably because we're registering WalletRegistry with the same contract type as is the implementation in the DI. Reverted.